cloud/amazon/elastic_beanstalk.py

```diff
@@ -0,0 +1,453 @@
+    except (boto.beanstalk.exception.InsufficientPrivileges,
+            boto.exception.BotoServerError) as error:
+        module.fail_json(
+            msg="Unable to terminate Environment '{0}': {1}".format(
+                environment_name, error
+            )
+        )
+    else:
+        changed = True
+        while retrieve_environment(module, beanstalk):
+            time.sleep(10)
+        environment = retrieve_environment(module, beanstalk)
+    return (changed, environment)
+
+def make_restarted(module, beanstalk):
+    """If the environment exists, restart its app server, and wait for it to come up."""
+
+    application_name = module.params.get("application_name")
+    environment_name = module.params.get("environment_name")
+    changed = False
+
+    if not retrieve_environment(module, beanstalk):
+        module.fail_json(msg="Environment '{0}' not found for Application '{1}'.".format(
+            environment_name,
+            application_name
+        ))
+
+    try:
+        beanstalk.restart_app_server(environment_name=environment_name)
+    except boto.exception.BotoServerError as error:
+        module.fail_json(msg="Unable to restart application server: '{0}'".format(error))
+    else:
+        changed = True
+        status = None
+        while status != "Ready":
+            time.sleep(10)
+            status = retrieve_environment(module, beanstalk)["Status"]
+        environment = retrieve_environment(module, beanstalk)
+    return (changed, environment)
+
+def main():
+    """Set arguments, perform Ansible module setup, and call state implementation functions."""
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            application_name=dict(required=True),
+            application_version=dict(),
+            environment_name=dict(
+                default="".join(
+                    random.choice(string.ascii_letters + string.digits) for __ in range(10)
+                )
+            ),
+            cname=dict(
+                default="".join(
+                    random.choice(string.ascii_letters + string.digits) for __ in range(10)
+                )
+            ),
+            application_s3_bucket=dict(),
+            application_s3_key=dict(),
+            solution_stack=dict(),
+            environment_options=dict(type="list"),
+            state=dict(choices=["present", "absent", "restarted"], required=True),
+            redeploy=dict(default=False, type="bool")
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[[
+            "configuration_template",
+            "environment_options"
+        ]]
+    )
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
+    aws_connect_kwargs.pop("validate_certs")
+    beanstalk = boto.connect_beanstalk(**aws_connect_kwargs)
```

Comment:
This does not properly respect a region specified from the get_aws_connection_info() helper ansible python module. Looks like it currently uses this connect_beanstalk which just connects to us-east-1. Switching to the boto.beanstalk.connect_to_region() method seems to have the same functionality along with region support.